### PR TITLE
Multi az

### PR DESCRIPTION
**Pull request by tmgstevens**
Draft PR for mutli AZ terraform. There's a merge error in there for some reason, which I'll go back at fix, but wanted to get this out there.

### CODE REVIEW

File: azure/vars.tf [hunk 1]
```diff
@@ -3,9 +3,9 @@ variable "region" {
   default     = "centralus"
 }
 
-variable "zone" {
+variable "availability_zone" {
   description = "Availability Zone"
-  default     = null
+  default     = []
```
[the remark this answers is not in the document]
Comment (vuldin) [reply azure/vars.tf:8]: Changing this variable name requires the following line to be updated: https://github.com/redpanda-data/deployment-automation/blob/main/gcp/provider.tf#L4

The immediate issue I see is that the `zone` parameter takes a single value, but `var.availability_zone` is a tuple. It looks like this is just used as the default zone, and any zone added to sub-resources will take precedence.

https://registry.terraform.io/providers/hashicorp/google/latest/docs/guides/provider_reference

**Review by vuldin — CHANGES_REQUESTED** [review]
I've tested on AWS and GCP, and found an issue with the GCP changes (comments below). I'll test on Azure soon.

File: azure/vars.tf (hunk 1 above)
[the remark this answers is not in the document]
Comment (tmgstevens) [reply azure/vars.tf:8]: Fixed the GCP issue.

**Comment by gene-redpanda** [comment]
@tmgstevens https://github.com/redpanda-data/deployment-automation/pull/151 I've updated this with the latest stuff from main and resolved the merge commits. LMK when you get a sec to give it a once over.

**Comment by tmgstevens** [comment]
Fixed by #165

**Pull request closed without merging**